skyrim-platform/src/platform_se/skyrim_platform/SkyrimPlatformProxy.h

```diff
@@ -1,7 +1,32 @@
 #pragma once
+#include "CallNativeApi.h"
 
 class SkyrimPlatformProxy
 {
 public:
-  static JsValue Attach(const JsValue& exports);
+  static JsValue Attach(
+    const JsValue& exports,
+    const std::function<CallNativeApi::NativeCallRequirements()>&
+      getNativeCallRequirements);
+};
+
+class JsFunctionArgumentsVectorImpl : public JsFunctionArguments
+{
+public:
+  JsFunctionArgumentsVectorImpl(const std::vector<JsValue>& arr_)
+    : arr(arr_)
+    , n(arr_.size())
+  {
+  }
+
+  size_t GetSize() const noexcept override { return n; }
+
+  const JsValue& operator[](size_t i) const noexcept override
+  {
+    return arr[i];
```

Comment:
oops. didn't see.

you should check size here and return undefined in case of bad index. like in JsFunctionArgumentsImpl.